public/app/plugins/datasource/zipkin/datasource.ts

```diff
@@ -24,7 +24,7 @@ export class ZipkinDatasource extends DataSourceApi<ZipkinQuery> {
     super(instanceSettings);
   }
 
-  query(options: DataQueryRequest<ZipkinQuery>): Observable<DataQueryResponse> {
```

Comment:
Why remove all the return types? I think we agreed that although TS can infer this, that it adds clarity?

Reply:
ohh I was not aware of that. I agree it adds clarity but I would enforce that "rule" (maybe we should add a rule) to public facing api. I like that ts can infer type for me so any time I change something I can see that it is breaking something or not without updating types.

Reply:
Enforcing to public API could make sense. Personally I prefer the return types explicitly because otherwise it can be a harder to get to the type in IDE and also if you have multiple return types it is easy to get to state where you have `: TypeA | TypeB` return type which is valid and the resulting compilation errors can be misleading of what is wrong.